eZ/Publish/SPI/Search/Content/IndexerGateway.php

```diff
@@ -40,6 +40,18 @@ public function getContentInSubtree(string $locationPath, int $iterationCount):
      */
     public function countContentInSubtree(string $locationPath): int;
 
+    /**
+     * @throws \Doctrine\DBAL\Exception
+     *
+     * @return iterable<int> list of Content IDs for each iteration
+     */
+    public function getContentWithContentTypeIdentifier(string $contentTypeIdentifier, int $iterationCount): iterable;
+
+    /**
+     * @throws \Doctrine\DBAL\Exception
+     */
+    public function countContentWithContentTypeIdentifier(string $contentTypeIdentifier): int;
```

Comment:
Are we actually ok with introducing new method into this class @alongosz? It is a contract.